### ISSUE TITLE
incorrect loss calculation

### ISSUE THREAD

**Issue by quantitative-technologies** [huggingface/transformers#13758]
## Environment info

- `transformers` version: 4.11.0.dev0
- Platform: Linux-5.4.104+-x86_64-with-Ubuntu-18.04-bionic
- Python version: 3.7.12
- PyTorch version (GPU?): 1.9.0+cu102 (False)
- Tensorflow version (GPU?): 2.6.0 (False)
- Flax version (CPU?/GPU?/TPU?): not installed (NA)
- Jax version: not installed
- JaxLib version: not installed
- Using GPU in script?: No
- Using distributed or parallel set-up in script?: Yes

### Who can help

@sgugger

## Information

This bug report is based on reading the source code.

The training loss appears to be calculated incorrectly when the batch loss is either nan or infinity:

In trainer.py lines 1314--1318: 
```
if args.logging_nan_inf_filter and (torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step)):
    # if loss is nan or inf simply add the average of previous logged losses
    tr_loss += tr_loss / 1 + (self.state.global_step - self._globalstep_last_logged)
else:
    tr_loss += tr_loss_step
```

I believe a pair of parenthesis was forgotten, and it should be:
```
if args.logging_nan_inf_filter and (torch.isnan(tr_loss_step) or torch.isinf(tr_loss_step)):
    # if loss is nan or inf simply add the average of previous logged losses
    tr_loss += tr_loss / (1 + (self.state.global_step - self._globalstep_last_logged))
else:
    tr_loss += tr_loss_step
```

**Comment by sgugger** [comment]
Indeed, thanks a lot for reporting. I would normally suggest to let you write a PR to fix this but we have a release planned today and the fix will need to be inside, so I'll fix it and make you co-author of the commit.

**Comment by LysandreJik** [comment]
Fixed by #13760
